bot/exts/utilities/color.py

```diff
@@ -0,0 +1,330 @@
+                        description=ERROR_MSG.format(user_color=user_color),
+                    )
+                )
+        elif mode.lower() == "rgb":
+            rgb_color = self.tuple_create(user_color)
+        elif mode.lower() == "hsv":
+            hsv_temp = self.tuple_create(user_color)
+            rgb_color = self.hsv_to_rgb(hsv_temp)
+        elif mode.lower() == "hsl":
+            hsl_temp = self.tuple_create(user_color)
+            rgb_color = self.hsl_to_rgb(hsl_temp)
+        elif mode.lower() == "cmyk":
+            cmyk_temp = self.tuple_create(user_color)
+            rgb_color = self.cmyk_to_rgb(cmyk_temp)
+        elif mode.lower() == "name":
+            color_name, hex_color = self.match_color_name(user_color)
+            if "#" in hex_color:
+                rgb_color = ImageColor.getcolor(hex_color, "RGB")
+            else:
+                rgb_color = ImageColor.getcolor("#" + hex_color, "RGB")
+        else:
+            # mode is either None or an invalid code
+            if mode is None:
+                no_mode_embed = Embed(
+                    title="No 'mode' was passed, please define a color code.",
+                    description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                    color=Colours.soft_red,
+                )
+                await ctx.send(embed=no_mode_embed)
+                return
+            wrong_mode_embed = Embed(
+                title=f"The color code {mode} is not a valid option",
+                description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=wrong_mode_embed)
+            return
+
+        (r, g, b) = rgb_color
+        discord_rgb_int = int(f"{r:02x}{g:02x}{b:02x}", 16)
+        all_colors = self.get_color_fields(rgb_color)
+        hex_color = all_colors[1]["value"].replace("» hex ", "")
+        if color_name is None:
+            logger.debug(f"Find color name from hex color: {hex_color}")
+            color_name = self.match_color_hex(hex_color)
+
+        async with ctx.typing():
+            main_embed = Embed(
+                title=color_name,
+                description='(Approx..)',
+                color=discord_rgb_int,
+            )
+
+            file = await self.create_thumbnail_attachment(rgb_color)
+            main_embed.set_thumbnail(url="attachment://color.png")
+            fields = self.get_color_fields(rgb_color)
+
+            for field in fields:
+                main_embed.add_field(
+                    name=field['name'],
+                    value=field['value'],
+                    inline=False,
+                )
+
+            await ctx.send(file=file, embed=main_embed)
+
+    @staticmethod
+    async def create_thumbnail_attachment(color: str) -> File:
+        """Generate a thumbnail from `color`."""
+        thumbnail = Image.new("RGB", (100, 100), color=color)
```

Comment:
I cannot see the thumbnail if I set it to that:

![image](https://user-images.githubusercontent.com/42116429/134158197-eaed4cfa-dc6d-4f6e-8fa2-f72d566df083.png)

I believe that is the size in pixels for the new image.

Reply:
Yes I believe it is, the smaller the image the better. Could be good to play around with different sizes, the smaller the better.

Reply:
Actually, I inspected a thumbnail image of some random embed, and found out that it's size is 80x80, so I just thought for larger monitors it maybe 100x100 (not sure though), and would be alright to keep...

But yeah, the smaller the better..

Reply:
I have a few examples prepped:

Original 100 x 100:
![image](https://user-images.githubusercontent.com/42116429/134454569-24a7b6da-c6c7-4178-b074-5d2fa951b006.png)

80 x 80:
![image](https://user-images.githubusercontent.com/42116429/134454591-0a7297ff-a9a1-49fb-8d7c-fefe4d79ddb2.png)

60 x 60:
![image](https://user-images.githubusercontent.com/42116429/134454619-c3435de3-f977-4815-b8cc-a1605ade3375.png)

40 x 40:
![image](https://user-images.githubusercontent.com/42116429/134454645-b2a5b71f-f762-439c-8053-fb68dfaabbc1.png)

20 x 20:
![image](https://user-images.githubusercontent.com/42116429/134454659-cb3cb9a3-5711-4d8d-933e-d3bef9e1b8f6.png)

Reply:
> Yes I believe it is, the smaller the image the better. Could be good to play around with different sizes, the smaller the better.

Although I'd argue this could be a premature optimisation.